datacube/api/query.py

```diff
@@ -33,15 +33,23 @@
 _LOG = logging.getLogger(__name__)
 
 
-GroupBy = collections.namedtuple('GroupBy', ['dimension', 'group_by_func', 'units', 'sort_key'])
-
 FLOAT_TOLERANCE = 0.0000001  # TODO: For DB query, use some sort of 'contains' query, rather than range overlap.
 SPATIAL_KEYS = ('latitude', 'lat', 'y', 'longitude', 'lon', 'long', 'x')
 CRS_KEYS = ('crs', 'coordinate_reference_system')
 OTHER_KEYS = ('measurements', 'group_by', 'output_crs', 'resolution', 'set_nan', 'product', 'geopolygon', 'like',
               'source_filter')
 
 
+class GroupBy:
+    """ Information needed to group datasets. """
+
+    def __init__(self, dimension, group_by_func, attrs, sort_key):
+        self.dimension = dimension
+        self.group_by_func = group_by_func
+        self.attrs = attrs
+        self.sort_key = sort_key
```

Comment:
If you going to replace tuple with object then might as well have `__call__(self, ds)` instead of `.group_by_func`, I would rename `.group_by_func` into `._impl` and call it from `__call__`. Maybe allow default that would return `id(ds)` to not group at all (that's assuming we don't put the same dataset twice into one time slice)